soroban-env-host/src/host.rs

```diff
@@ -3003,6 +3054,13 @@ impl Host {
     {
         f(self.0.budget.clone())
     }
+
+    pub fn retrieve_entry_with_lifetime(
```

Comment:
It seems like this function can be moved to "testutils". It also should be `pub(crate)`, not `pub`

Reply:
If I move this to testutils I won't be able to access the storage which is private. Should I keep this as is or add a testutils borrower within the host mod and call it from the testutils?